solr/solrj/src/java/org/apache/solr/common/cloud/DocCollection.java

```diff
@@ -383,10 +382,12 @@ public boolean equals(Object that) {
     return super.equals(that) && Objects.equals(this.name, other.name) && this.znodeVersion == other.znodeVersion;
   }
 
-//  @Override
-//  public int hashCode() {
-//    throw new UnsupportedOperationException("TODO unimplemented DocCollection.hashCode");
-//  }
+  @Override
+  public int hashCode() {
+    return Objects.hash(super.hashCode(), znodeVersion, name, replicationFactor,
+            numNrtReplicas, numTlogReplicas, numPullReplicas, maxShardsPerNode,
+            autoAddReplicas, policy, readOnly);
```

Comment:
The superclass (`ZkNodeProps`) `hashCode()` hashes the `props` map, so already takes care of producing a hash value based on `replicationFactor`, `numNrtReplicas`, `numTlogReplicas`, `numPullReplicas`, `maxShardsPerNode`, `autoAddReplicas`, `policy` and `readOnly` that were extracted from the map. These do not have to be added here again but it's **not** incorrect to have them.

Reply:
Fixed zkNodeProps to return the hash of the propmap. I think for this, the name and znodeversion are enough (not even adding returning super.hashCode) in the interests of speed.